contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,378 @@
+
+            require(
+                auctions_[i].minIncreaseFactor > 0,
+                "FeralfileEnglishAuction: auction min increase factor should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseAmount > 0,
+                "FeralfileEnglishAuction: auction min increase amount should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minPrice > 0,
+                "FeralfileEnglishAuction: auction min price should be greater than 0"
+            );
+
+            auctions[auctions_[i].id] = auctions_[i];
+        }
+    }
+
+    function listAuctionLatestStatus(
+        bytes32[] memory aucIds_
+    ) external view returns (LatestAuctionStatus[] memory) {
+        LatestAuctionStatus[] memory results = new LatestAuctionStatus[](
+            aucIds_.length
+        );
+        for (uint i = 0; i < aucIds_.length; i++) {
+            // get the auction object
+            EnglishAuction memory auction_ = auctions[aucIds_[i]];
+            // get the last auction bid object
+            results[i].latestBid = auctionLatestBid[aucIds_[i]];
+            results[i].endAt = auction_.endAt;
+            results[i].isSettled = auction_.isSettled;
+        }
+        return results;
+    }
+
+    function isValidBidRequest(
+        bytes32 aucId_,
+        address bidder_,
+        uint256 amount_
+    ) external view returns (bool) {
+        // get the auction object
+        EnglishAuction memory auction_ = auctions[aucId_];
+        // get the last auction bid object
+        Bid memory lastBid_ = auctionLatestBid[aucId_];
+        // verify the bid is valid
+        return
+            _isValidBidRequest(
+                auction_,
+                lastBid_,
+                bidder_,
+                amount_,
+                block.timestamp
+            );
+    }
+
+    function _isValidBidRequest(
+        EnglishAuction memory auction_,
+        Bid memory lastBid_,
+        address bidder_,
+        uint256 amount_,
+        uint256 timestamp_
+    ) internal pure returns (bool) {
+        // make sure the auction exist
+        require(
+            auction_.id.length > 0,
+            "FeralfileEnglishAuction: auction not found"
+        );
+        // check if auction is on going
+        require(
+            auction_.startAt <= timestamp_ && timestamp_ <= auction_.endAt,
+            "FeralfileEnglishAuction: auction not started or ended"
+        );
+        // check if auction is not settled
+        require(
+            !auction_.isSettled,
+            "FeralfileEnglishAuction: auction already settled"
+        );
+        // check if current winner not equal to current bidder (optional)
+        require(
+            lastBid_.bidder != bidder_,
+            "FeralfileEnglishAuction: bidder is the current winner"
+        );
+        // check bidding amount > highest bid amount
+        require(
+            amount_ > lastBid_.amount,
```

Comment:
The check based on minIncreaseAmount and minIncreaseFactor could cover this check, i think.